sympy/physics/units/tests/test_quantities.py

```diff
@@ -364,8 +364,7 @@ def test_factor_and_dimension():
 
     pH = -log(cH)
 
-    assert (1, volume/amount_of_substance) == Quantity._collect_factor_and_dimension(
-        exp(pH))
+    assert (1, Dimension(1)) == Quantity._collect_factor_and_dimension(exp(pH))
```

Comment:
This can't be right since cH doesn't round-trip. As I read it, `Quantity._collect_factor_and_dimension(cH)` will not be the same as `Quantity._collect_factor_and_dimension(exp(-pH))`...but they must be so.

Reply:
I think cH will be having a dimension of volume/amount_of_substance but exp(-pH) will be dimensionless since pH itself is dimensionless. I am saying so from the definition given [here](https://en.wikipedia.org/wiki/PH). However, if you still say that original tests should be restored, then I would surely restore it. :-)

Reply:
This is true and in this test, I think the wrong argument is being used with the log: the argument should be dimensionless, shouldn't it. But if you put garbage in (a dimensional unit) you should not get gold out ;-) 

But I don't understand the interaction of Quantities and Dimensions like `amount_of_substance/volume` and `mole/liter`. I get 

```
>>> m = amount_of_substance/volume
>>> m/(mole/liter)
Dimension(amount_of_substance/volume)
>>> m/hour
Dimension(amount_of_substance/volume)
```

Reply:
Ahh yes, you are right regarding units of `exp(-pH)`

In reply on the issue page about how arguments are made dimensionless the author writes:

`log(2*foot)` is made dimensionless as follows:

1. First `2*foot` is checked whether it's dimensionless. Clearly it's not so we will move to step 2.
By filtering out the 2 and foot from * using a pre-defined method in mul class.
1. Then 2 is checked whether it's dimensionless. Clearly, it's not so it is kept unchanged.
foot is checked whether it's dimensionless. Since, it is dimensionless, we convert this foot to dimensionless constant that is 1.
1. Then 2 and 1 are multiplied.
1. Finally log(2) is taken.

I could have replaced foot by 1000 or some other number but choosing 1 assumes that the user wants to take the log(2 foot) that is he expects, log(2). Thus, scale remains unchanged.

I am -1 to this idea. Units cannot be ignored. If one writes `log(2*second/hour)` the argument has no dimension and the normal expansion of the log would give `log(2) + log(second) - log(hour)`. If this is simplified so all units are "standard' then the correct numerical value will be obtained. Anything that ignores units will be wrong.

I am not sure that this is what this PR does, however:

```
>>> log(2*foot)
log(2*foot)
```

Reply:
Quite right, I will work more on this. By the way, it's been awesome to contribute to sympy. Thanks a lot for considering my views and giving me feedbacks regularly. It's so friendly to be here :-)
I will check on my local system that how my modifications respond to `log(2*foot)`.